src/components/Welcome.js

```diff
@@ -1,7 +1,35 @@
-import React from 'react';
+import React, { useState, useContext } from 'react';
 import { Link } from 'react-router-dom';
+import { ListContext } from '../context/ListContext';
 
 export default function Welcome() {
+  const [userInputToken, setUserInputToken] = useState('');
+
+  const currentList = useContext(ListContext);
+
+  const handleChange = (e) => {
+    setUserInputToken(e.target.value);
+  };
+
+  const handleJoinList = (e) => {
+    e.preventDefault();
+    currentList.itemsRef
+      .where('userToken', '==', userInputToken)
+      .limit(1)
+      .get()
+      .then(function (querySnapshot) {
+        if (querySnapshot.empty) {
+          alert('User Token Does Not Exist.Please Try Again');
+        } else {
+          localStorage.setItem('tcl13-token', userInputToken);
```

Comment:
@indiragp sorry to confuse you but this morning we updated the Welcome.js in this branch so we could all useContext. In doing that, we removed the unnecessary forEach. Earlier, we were wondering if we could just submit one PR from this branch.